### PR TITLE
Add tracing plug

### PR DESCRIPTION
**Pull request by milica-nerlovic**
I've moved the logic to the plug itself since it's closely connected to the conn. Let me know if you feel we should arrange things differently.

### CODE REVIEW

File: mix.exs [hunk 1]
```diff
@@ -11,18 +11,16 @@ defmodule Traceman.Mixfile do
...
     ]
   end
 
-  # Run "mix help deps" to learn about dependencies.
   defp deps do
     [
-      # {:dep_from_hexpm, "~> 0.3.0"},
-      # {:dep_from_git, git: "https://github.com/elixir-lang/my_dep.git", tag: "0.1.0"},
+      {:cowboy, "~> 2.0"},
+      {:plug, "~> 1.0"}
```
Comment (shiroyasha) [inline mix.exs]: Would be nice if the service could define these, I wonder if there is a pattern for that.

File: lib/traceman/plugs/trace_headers.ex [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        # if there is no such header, we don't inject anything into the
+        # just pass it to the next iteration unchanged
+
+        result
+      end
+    end)
+
+    Logger.info inspect(metadata)
+
+    assign conn, :metadata, metadata
```
Comment (shiroyasha) [inline lib/traceman/plugs/trace_headers.ex]: Maybe we should call this something more specific, like `tracing_headers` ?

File: lib/traceman.ex [hunk 3]
```diff
@@ -1,4 +1,5 @@
 defmodule Traceman do
+  require Logger
```
Comment (shiroyasha) [inline lib/traceman.ex:2]: Is this necessary?

File: lib/traceman/plugs/trace_headers.ex [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+        Map.put(result, header, hd(http_headers))
+      else
+        # if there is no such header, we don't inject anything into the
+        # just pass it to the next iteration unchanged
+
+        result
+      end
+    end)
+
+    Logger.info inspect(metadata)
```
Comment (shiroyasha) [inline lib/traceman/plugs/trace_headers.ex]: Lets remove the logger.

File: mix.exs (hunk 1 above)
Comment (milica-nerlovic) [reply mix.exs]: I guess. The tests would not work, but I can check if it would compile correctly.

Comment (shiroyasha) [reply mix.exs]: Maybe somthing like, `{:cowboy, "~> 2.0", only: test}` would work? (never tried this)

Comment (milica-nerlovic) [reply mix.exs]: It seems to work! 👍 I'll check how it behaves when added as a dependency.

**Pull request merged**